ethereum/circuits/lib/src/account_with_storage.nr

```diff
@@ -49,42 +52,76 @@ struct StorageWithinBlock<N> {
     values: [Bytes32; N],
 }
 
+type ProofInputSerialized<LEN> = [Field; LEN];
+
 impl Eq for StorageWithinBlock<1> {
     fn eq(self, other: Self) -> bool {
         (self.block_hash == other.block_hash) & (self.account == other.account) & (self.values[0] == other.values[0])
     }
 }
 
+fn assert_storage_key_equals(
+    storage_key: Bytes32,
+    storage_key_hash: [u8; MAX_PREFIXED_KEY_NIBBLE_LEN]
+) {
+    let storage_key_hash_fragment = Fragment::new(
+        MAX_PREFIXED_KEY_NIBBLE_LEN - HASH_LEN,
+        HASH_LEN,
+        storage_key_hash
+    );
+    let other_storage_key_hash_fragment = Fragment::from_array(keccak256(storage_key, BYTES32_LENGTH as u32));
+    assert(
+        storage_key_hash_fragment.eq(other_storage_key_hash_fragment), "Storage key does not match the argument"
+    );
+}
+
+fn get_storage_value(rlp_encoded_value: [u8; MAX_STORAGE_VALUE_LEN]) -> [u8; MAX_STORAGE_VALUE_LEN] {
+    let mut storage_value = rlp_encoded_value;
+    let rlp_value_len = right_pad(rlp_encoded_value).len();
```

Comment:
left_padded_len

Reply:
I rewrote this function, now it uses Fragments and I think this name is outdated